src/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareParameters.cs

```diff
@@ -0,0 +1,215 @@
+using CarbonAware.Model;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CarbonAware.Aggregators.CarbonAware;
+
+public class CarbonAwareParametersBaseDTO
+{
+    virtual public string[]? MultipleLocations { get; set; }
+    virtual public DateTimeOffset? Start { get; set; }
+    virtual public DateTimeOffset? End { get; set; }
+
+    public Dictionary<string, string> GetDisplayNameMap()
+    {
+        var mapping = new Dictionary<string, string>();
+        foreach (var DTOproperty in this.GetType().GetProperties())
+        {
+            BindPropertyAttribute? customAttribute = (BindPropertyAttribute?) Attribute.GetCustomAttribute(DTOproperty, typeof(BindPropertyAttribute));
+            if (customAttribute != null && customAttribute.Name != null)
+            {
+                mapping.Add(DTOproperty.Name, customAttribute.Name);
+            }
+        }
+        return mapping;
+    }
+}
+
+public class CarbonAwareParameters
+{
```

Comment:
Just for my knowledge, how do we determine the property naming conventions? I see here it is preceded by -'_' (_multipleLocations) but in the other class it is Pascal case (MulyipleLocations)